solutions/usaco-622.mdx

```diff
@@ -0,0 +1,81 @@
+---
+id: usaco-622
+source: USACO Gold 2016 February
+title: Circular Barn Revisited
+author: Chuyang Wang
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_cbarn2_gold_feb16.html)
+
+## Explanation
+
+The problem asks for the minimum amount of distance the cows need to travel for a circular barn. However, if we fix the position of the first door, the problem can be reduced to a linear barn $\texttt{rooms}[1..n]$ where we place the first door at the beginning of our linear barn. In such case, we can define $\texttt{dp}[i][j]$ as the sum of distance to fill the rooms $[j, n]$ if we place the $i$-th door at $j$. All values of the $\texttt{dp}$ array are initialized with infinity. We further set $\texttt{dp}[0][n+1] = 0$ because with no doors the only case where the distance is zero is to fill no room. Our transitions would then be
+
+$\texttt{dp}[i][j] = min(\texttt{dp}[i][j], \sum_{p=j+1}^{n+1} (p-j-1) \cdot \texttt{rooms}[p-1] + \texttt{dp}[i-1][p])$,
```

Comment:
what does this mean in normalspeak

Reply:
added an explanation for the equation after it